apis/v0/jwt.go

```diff
@@ -102,3 +108,45 @@ func ClearJustifications(t jwt.Token) error {
 
 	return t.Remove(jwtJustificationsKey)
 }
+
+// ToJSON converts the token into json with justification claims seperated from other claims.
+func ToJSON(ctx context.Context, t jwt.Token) ([]byte, error) {
+	return marshal(ctx, t, "\n---", func(v any) ([]byte, error) { return json.MarshalIndent(v, "", "  ") })
+}
+
+// ToYAML converts the token into yaml with justification claims seperated from other claims.
+func ToYAML(ctx context.Context, t jwt.Token) ([]byte, error) {
+	return marshal(ctx, t, "---", func(v any) ([]byte, error) { return yaml.Marshal(v) })
+}
+
+// marshal converts the token into byte array using the given malshal function
+// with justification claims seperated from other claims using the given seperator.
+func marshal(ctx context.Context, t jwt.Token, sep string, fn marshalFn) ([]byte, error) {
+	if t == nil {
+		return nil, fmt.Errorf("token cannot be nil")
+	}
+
+	claimsMap, err := t.AsMap(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert token into map: %w", err)
+	}
+	delete(claimsMap, jwtJustificationsKey)
+	claimsOut, err := fn(claimsMap)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal token as json: %w", err)
+	}
+
+	justs, err := GetJustifications(t)
+	if err != nil {
+		return nil, fmt.Errorf("failed to extract justifications: %w", err)
+	}
+	justsOut, err := fn(justs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal justifications as json: %w", err)
+	}
```

Comment:
Marshal the justifications will end up some unnecessary info:

- marshal result `category: explanation, value: i need access` vs.
- better human readable output `explanation: i need access`